loadtest/store/memstore/random.go

```diff
@@ -350,24 +349,24 @@ func (s *MemStore) RandomCategory(teamID string) (model.SidebarCategoryWithChann
 		return model.SidebarCategoryWithChannels{}, err
 	}
 
-	category := *teamCat[key.(string)]
+	category := *teamCat[key]
 	tmp := make([]string, len(category.Channels))
 	copy(tmp, category.Channels)
 	category.Channels = tmp
 	return category, nil
 }
 
-func pickRandomKeyFromMap(m interface{}) (interface{}, error) {
-	val := reflect.ValueOf(m)
-	if val.Kind() != reflect.Map {
-		return nil, errors.New("memstore: not a map")
+func pickRandomKeyFromMap[K comparable, V any](m map[K]V) (K, error) {
+	var def K
+	if len(m) == 0 {
+		return def, ErrEmptyMap
```

Comment:
Purely out of curiosity, can't you initialize K in place?

Reply:
I tried to do that but couldn't think of anything at the moment. I can't do `K{}` obviously, and nor can I do `new(K)` because we are returning K not *K. What else am I missing?

Reply:
Looks like you can do `*new(K)`, jeez :roll_eyes:

Reply:
oh dear ..